src/hangar/checkout.py

```diff
@@ -127,6 +134,236 @@ def __verify_checkout_alive(self):
+
+        Get sample from multiple arraysets (returns namedtuple of arrays, field
+        names = arrayset names)
+
+            >>> dset[('foo', 'bar', 'baz'), '1']
+            ArraysetData(foo=array([1]), bar=array([11]), baz=array([111]))
+
+        Get multiple samples from multiple arraysets(returns list of namedtuple
+        of array sorted in input key order, field names = arrayset names)
+
+            >>> dset[('foo', 'bar'), ('1', '2')]
+            [ArraysetData(foo=array([1]), bar=array([11])),
+             ArraysetData(foo=array([2]), bar=array([22]))]
+
+        Get samples from all arraysets (shortcut syntax)
+
+            >>> out = dset[:, ('1', '2')]
+            >>> out = dset[..., ('1', '2')]
+            >>> out
+            [ArraysetData(foo=array([1]), bar=array([11]), baz=array([111])),
+             ArraysetData(foo=array([2]), bar=array([22]), baz=array([222]))]
+
+            >>> out = dset[:, '1']
+            >>> out = dset[..., '1']
+            >>> out
+            ArraysetData(foo=array([1]), bar=array([11]), baz=array([111]))
+
+        Parameters
+        ----------
+        index
+            Please see detailed explanation above for full options. Hard coded
+            options are the order to specification.
+
+            The first element (or collection) specified must be ``str`` type and
+            correspond to an arrayset name(s). Alternativly the Ellipsis operator
+            (``...``) or unbounded slice operator (``:`` <==> ``slice(None)``) can
+            be used to indicate "select all" behavior.
+
+            If a second element (or collection) is present, the keys correspond to
+            sample names present within (all) the specified arraysets. If a key is
+            not present in even on arrayset, the entire ``get`` operation will
+            abort with ``KeyError``. If desired, the same selection syntax can be
+            used with the :meth:`~hangar.checkout.ReaderCheckout.get` method, which
+            will not Error in these situations, but simply return ``None`` values
+            in the appropriate position for keys which do not exist.
+
+        Returns
+        -------
+        Arrayset
+            single arrayset parameter, no samples specified
+
+        np.ndarray
+            Single arrayset specified, single sample key specified
+
+        List[np.ndarray]
+            Single arrayset, multiple samples array data for each sample is
+            returned in same order sample keys are recieved.
+
+        List[NamedTuple[*np.ndarray]]
+            Multiple arraysets, multiple samples. Each arrayset's name is used
+            as a field in the NamedTuple elements, each NamedTuple contains
+            arrays stored in each arrayset via a common sample key. Each sample
+            key is returned values as an individual element in the
+            List. The sample order is returned in the same order it wasw recieved.
+
```

Comment:
It might be nice to have that in some cases, but since there isn't anyway to set some modifiable `flag` during dict style access it would have to be permenant. Two levels of indexing to pull elements which could be stored sequentially in a list would drive me nuts...

Reply:
No, what I am saying is, It is already returning a NamedTuple in the case `dset[('foo', 'bar', 'baz'), '1']` which is not documented in the `Returns` field. Or Am I missing something here?